### PR TITLE
adds client lib from chart-operator

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2737.

### CODE REVIEW

File: apprclient.go [hunk 1]
```diff
@@ -1 +1,167 @@
+// Package appr holds the client code required to interact with a CNR backend.
```
Comment (rossf7) [inline apprclient.go]: apprclient

File: apprclient.go [hunk 2]
```diff
@@ -1 +1,167 @@
...
+	organization string
+}
+
+// New creates a new configured appr client.
+func New(config Config) (*Client, error) {
+	if config.Fs == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Fs must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.logger must not be empty", config)
```
Comment (rossf7) [inline apprclient.go]: Logger

**Review by rossf7 — APPROVED** [review]
LGTM just a couple of minor comments.

**Review by fgimenez — APPROVED** [review]
LGTM

File: apprclient.go (hunk 2 above)
Comment (xh3b4sd) [reply apprclient.go]: Good catch. Was the same for the settings. Fixed.

File: apprclient.go (hunk 1 above)
Comment (xh3b4sd) [reply apprclient.go]: Cool. Done.

**Pull request merged**